x-pack/plugins/security/server/authorization/privileges/feature_privilege_builder/cases.ts

```diff
@@ -15,6 +15,9 @@ import { BaseFeaturePrivilegeBuilder } from './feature_privilege_builder';
 const readOperations: string[] = [
   'getCase',
   'findCases',
+  'getComment',
+  'getAllComments',
+  'findComments',
```

Comment:
Do we really need separate feature privileges for `getComment`, `getAllComments`, _and_ `findComments`? Or, to rephrase: will there ever be a user who can get a single comment but not all comments? Will there ever be a user who can get comments but not find comments?

If not, we should probably consider removing `getAllComments` and `findComments`, and instead just use `getComment` to authorize for this operation. Every separate action/operation in the feature privileges definition has a performance impact for authorization checks against Elasticsearch.

Reply:
Thanks for the comment @jportner ! Great points. @legrego when we had initially talked a while ago my interpretation was that we needed an operation per route. I'm happy to remove the extra ones though.

Also I didn't realize that having separate actions/operations would result in worse performance. So the larger these arrays are the worse it'll be?

Reply:
My recommendation for an operation-per-route was based on following our current model for general saved objects authorization:

https://github.com/elastic/kibana/blob/6ecffcc57c9a5b0328257cd25b737140a8471dda/x-pack/plugins/security/server/authorization/privileges/feature_privilege_builder/saved_object.ts#L13-L27

---

That said, we don't have an actual need for this level of granularity, since our privilege system only distinguishes between `all` and `read` today. At one point this setup simplified our audit logging, but that's only partially true now.

> Every separate action/operation in the feature privileges definition has a performance impact for authorization checks against Elasticsearch.

> Also I didn't realize that having separate actions/operations would result in worse performance. So the larger these arrays are the worse it'll be?


The performance issues are mostly hypothetical. There's an upper bound to everything, but we don't know what this is for our privileges yet. The payload we exchange with Elasticsearch to describe each or our privileges is quite large.

The work that ES does to perform the authorization checks is opaque to me - we again haven't run into issues yet, but I want to make sure we don't end up with a surprise that's hard to recover from.

----

So given all of that, I don't have a strong opinion for Cases RBAC either way. If you're comfortable reducing this down to a smaller set (and diverging from the way we currently do things), then I have no objections. If there is too much momentum around the current approach to refactor this now, then I'm ok to leave it as-is.

Reply:
Thanks for the response @legrego. Makes sense. It's pretty easy to do the changes @jportner asked for, so I'll refactor it down to just a `getComment` and `deleteComment` 👍  The way we've implemented the audit logging we can still log  whether it was a `find comment`/`get comment`/`get all comments` etc. So we should be good there.